src/tests/Common/CLRTest.Jit.targets

```diff
@@ -365,6 +372,139 @@ set DOTNET_JitPath=%CORE_ROOT%\superpmi-shim-collector.dll
+      <_IlasmRoundTripScriptText><![CDATA[
+import os
+import shutil
+import subprocess
+import sys
+import glob
+import struct
+import platform
+
+def is_managed_assembly(file):
+    file_size = os.path.getsize(file)
+
+    if file_size < 64:
+        return False
+
+    with open(file, "rb") as f:
+        f.seek(0x3c)
+
+        # PE Header starts @ 0x3C (60). Its a 4 byte header.
+        peHeaderPointer = struct.unpack('I', f.read(4))[0]
+        if peHeaderPointer == 0:
+            peHeaderPointer = 0x80
+
+        # Ensure there is at least enough room for the following structures:
+        #     24 byte PE Signature & Header
+        #     28 byte Standard Fields         (24 bytes for PE32+)
+        #     68 byte NT Fields               (88 bytes for PE32+)
+        # >= 128 byte Data Dictionary Table
+        if peHeaderPointer > (file_size - 256):
+            return False
+
+        # Check the PE signature.  Should equal 'PE\0\0'.
+        f.seek(peHeaderPointer)
+        peHeaderSignature = struct.unpack('I', f.read(4))[0]
+        if peHeaderSignature != 0x00004550:
+            return False
+
+        # skip over the PEHeader fields
+        f.seek(f.tell() + 20)
+
+        PE32 = 0x10b
+        PE32Plus = 0x20b
+
+        # Read PE magic number from Standard Fields to determine format.
+        peFormat = struct.unpack('H', f.read(2))[0]
+        if peFormat != PE32 and peFormat != PE32Plus:
+            return False
+
+        # Read the 15th Data Dictionary RVA field which contains the CLI header RVA.
+        # When this is non-zero then the file contains CLI data otherwise not.
+        dataDictionaryStart = (peHeaderPointer + (232 if peFormat == PE32 else 248)) & 0xffff
+        f.seek(dataDictionaryStart)
+
+        cliHeaderRva = struct.unpack('I', f.read(4))[0]
+        if cliHeaderRva == 0:
+            return False
+
+        return True
+
+print("")
+print("ILASM RoundTrips")
+
+if not os.path.exists("IL-RT"):
+  os.mkdir("IL-RT")
+
+if "arm" in platform.machine():
+  # TODO: Fix these roundtrip assemblies from failing on ARM64 when executing.
+  assembliesToIgnore = [
+    "value_numbering_checked_arithmetic_with_constants_ro.dll",
+    "ldc_mul_ovf_i4.dll",
+    "overflow02_mul.dll",
+    "b25815.dll",
+    "b71869.dll"
+  ]
+else:
+  assembliesToIgnore = []
+
+for inputAssemblyName in glob.glob("*.dll"):
+  if (not inputAssemblyName.endswith("$(AssemblyName).dll")) and is_managed_assembly(inputAssemblyName):
+    if inputAssemblyName not in assembliesToIgnore:
+      print("ILASM RoundTrip for " + inputAssemblyName)
+
+      ilasmSwitches = ""
+
+      # HACK: The 'poison.csproj' is the only project that needs to be built with the debug flag.
+      #       We do this here because we cannot easily determine whether or not it was actually built with the flag or not.
+      if inputAssemblyName.endswith("poison.dll"):
+        ilasmSwitches = "-DEBUG"
```

Comment:
See `_IlasmSwitches`

Reply:
`_IlasmSwitches` will not work here unfortunately. I tried this. The problem is that `_IlasmSwitches` refers to the merged test assembly and not `poison.csproj` which specifies the `DebugType`.

Reply:
This seems like a big problem.  There are lots of other tests that specify DebugType.

This is getting beyond my msbuild experience, but those properties are all around.  We may need to persist them.

Reply:
That's true. Only thing I can think of is detecting if the assembly was marked as optimized or not and go based on that.

Reply:
Trying to reconstruct everything seems like a losing battle.  We "just" need to figure out how to get the information from msbuild to test-time.